sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -322,11 +322,126 @@ q.submit([&](sycl::handler& cgh) {
 });
 q.wait();
 ```
+
+=== Load/Store Using Annotated Pointers
+These are overloads of existing `joint_matrix_load` and
+`joint_matrix_store` that take
+`sycl::ext::oneapi::experimental::annotated_ptr` as argument instead
+of `sycl::multi_ptr`.
+The property list associated with the `annotated_ptr` argument
+represents the compile-time constant properties for cache control included
+in the SYCL extenion
+link:../../proposed/sycl_ext_intel_cache_controls.asciidoc[sycl_ext_intel_cache_controls]
+
+
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+
+// Only available when std::is_same_v<T1, std::remove_const_t<T2>>
+template <typename Group, typename T1, typename T2,
+          size_t Rows, size_t Cols,
+          typename PropertyListT>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, T1, use::accumulator, Rows, Cols, layout::dynamic> &res,
+    ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> src,
+    size_t stride, layout Layout);
+
+// Only available when Layout != layout::dynamic
+// and when std::is_same_v<T1, std::remove_const_t<T2>>
+template <typename Group, typename T1, typename T2,
+          size_t Rows, size_t Cols, use Use, layout Layout,
+          typename PropertyListT>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, T1, Use, Rows, Cols, Layout> &res,
+    ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> src,
+    size_t stride);
+
+} // namespace sycl::ext::intel::experimental::matrix
+```
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+
+template <typename Group, typename T1, typename T2, size_t Rows, size_t Cols,
+          typename PropertyListT>
+void joint_matrix_store(Group g,
+   const joint_matrix<Group, T1, use::accumulator, Rows, Cols, layout::dynamic> &res,
+   ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> dest,
+   size_t stride, layout Layout);
+
+// Only available when Layout != layout::dynamic
+template <typename Group, typename T1, typename T2,
+          size_t Rows, size_t Cols, use Use, layout Layout,
+          typename PropertyListT>
+void joint_matrix_store(Group g,
+    joint_matrix<Group, T1, Use, Rows, Cols, Layout> &res,
+    ext::oneapi::experimental::annotated_ptr<T2, PropertyListT> src,
+    size_t stride);
+
+} // namespace sycl::ext::intel::experimental::matrix
+```
+==== Example
+
+```c++
+using syclex = sycl::ext::oneapi::experimental;
+using syclintelex = sycl::ext::intel::experimental;
+
+auto aptr = annotated_ptr(ptr, syclex::properties{syclintelex::cache_control_read<cache_control_read_type::cached, cache_level::L2>});
```

Comment:
Keep an eye on #11584, which is changing the syntax for these cache controls.  You will need to update this example when that PR stabilizes.